### PR TITLE
docs(messaging): update notification example for react-navigation v6

### PR DESCRIPTION
**Pull request by anhtuank7c**
You cannot use `useNavigation` hooks from outside of `NavigationContainer` The properly way is to implement `linking` props of `NavigationContainer`. Reference document: https://reactnavigation.org/docs/navigation-container#linkingsubscribe

### Description
The previous document was invalid for at least React Navigation v6.

### Related issues
React Navigation won't let you use `useNavigation` hooks from the outside of `NavigationContainer` scope.

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [x] `Android`
  - [x] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [x] Yes
  - [ ] No

### Test Plan

Need a real device (for iOS) and emulator/real device for Android to test.
Use FCM to push a notification with these data: `navigationId: required` and `postId: optional`

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Review by mikehardy — CHANGES_REQUESTED** [review]
Hi there - this is no longer a minimal example, what you propose adds a screen, brings in linking and all sorts of things. The previous example was much much simpler and just showed the elements of the integration

**Comment by anhtuank7c** [comment]
@mikehardy I've removed the PostScreen component to simplify the example. The linking configuration is required to make notification interaction work. Previous example doesn't work.

**Comment by mikehardy** [comment]
Apologies for the delay @anhtuank7c ! Thanks for posting this. I can merge it but not without the CLA signed, can you please follow the details link (should be https://cla-assistant.io/invertase/react-native-firebase?pullRequest=7439) and sign the CLA?

Thank you!

**Review by mikehardy — APPROVED** [review]
OK - this looks great now - I really appreciate the contribution and hopefully it will help people

Just pending the CLA signing now - follow the details link on that CI check or in the CLA-bot notification and we should be able to get this merged after signing

Thanks!

File: docs/messaging/notifications.md [hunk 1]
```diff
@@ -112,61 +112,90 @@ we can set an initial route when the app is opened from a quit state, and push t
...
-  }, []);
-
-  if (loading) {
+function buildDeepLinkFromNotificationData(data): string | null {
+  const navigationId = data?.navigationId;
+  if (!NAVIGATION_IDS.includes(navigationId)) {
+    console.warn('Unverified navigationId', navigationId)
     return null;
   }
+  if (navigationId === 'home) {
```
Comment (maximownrock) [inline docs/messaging/notifications.md:129]: 'home'

Comment (mikehardy) [inline docs/messaging/notifications.md:129]:
```suggestion
  if (navigationId === 'home') {
```

**Review by mikehardy — APPROVED** [review]

**Comment by anhtuank7c** [comment]
@mikehardy is there anything left that prevent us to merge this PR?

**Comment by mikehardy** [comment]
Oh! Sorry this sat here. I think it's good to go, thank you

**Review by mikehardy — APPROVED** [review]

**Pull request merged**